client/asset/btc/btc.go

```diff
@@ -1385,6 +1387,96 @@ func (btc *baseWallet) Info() *asset.WalletInfo {
 	return btc.walletInfo
 }
 
+func (btc *baseWallet) startTxHistoryDB() error {
+	dir, err := os.Open(btc.walletDir)
+	if err != nil {
+		return fmt.Errorf("error opening wallet directory: %w", err)
+	}
+	defer dir.Close()
+
+	entries, err := dir.Readdir(0)
+	if err != nil {
+		return fmt.Errorf("error reading wallet directory: %w", err)
+	}
+
+	pattern := regexp.MustCompile(`^txhistory-(.+)\.db$`)
+
+	var dbPath string
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+
+		match := pattern.FindStringSubmatch(entry.Name())
+		if match == nil {
+			continue
+		}
+
+		address := match[1]
+		owns, err := btc.OwnsDepositAddress(address)
+		if err != nil {
+			btc.log.Errorf("Error checking if wallet owns deposit address %s: %w", address, err)
+			continue
+		}
+		if owns {
+			btc.log.Infof("Using tx history db %s", entry.Name())
+			dbPath = filepath.Join(btc.walletDir, entry.Name())
+			break
+		}
```

Comment:
Clever scheme, but I'll note that it may be fragile in some rare edge cases. If the user is for some reason forced to rebuild the transaction db later on after some use, perhaps because of file corruption, the address associated with the db might be from a higher index. If they later restore from seed, the synchronization will have just started by this point, and that address won't be known to the wallet yet, so a new db file will be created. Not a big deal at all. Just an observation.